app/org/sagebionetworks/bridge/services/ParticipantService.java

```diff
@@ -347,9 +360,32 @@ public IdentifierHolder createParticipant(Study study, Set<Roles> callerRoles, S
         if (sendEmailVerification && !study.isAutoVerificationEmailSuppressed()) {
             accountWorkflowService.sendEmailVerificationToken(study, accountId, account.getEmail());
         }
+
+        // If you create an account with a phone number, this opts the phone number in to receiving SMS. We do this
+        // _before_ phone verification / sign-in, because we need to opt the user in to SMS in order to send phone
+        // verification / sign-in.
+        Phone phone = account.getPhone();
+        if (phone != null) {
+            // Check if phone number is opted out.
+            CheckIfPhoneNumberIsOptedOutRequest checkRequest = new CheckIfPhoneNumberIsOptedOutRequest()
+                    .withPhoneNumber(phone.getNumber());
+            CheckIfPhoneNumberIsOptedOutResult checkResult = snsClient.checkIfPhoneNumberIsOptedOut(checkRequest);
+
+            if (Boolean.TRUE.equals(checkResult.isOptedOut())) {
+                LOG.info("Opting in user " + accountId + " for SMS messages");
+
+                // User was previously opted out. They created a new account (almost certainly in a new study). We need
+                // to opt them back in. Note that according to AWS, this can only be done once every 30 days to prevent
+                // abuse.
+                OptInPhoneNumberRequest optInRequest = new OptInPhoneNumberRequest().withPhoneNumber(
+                        phone.getNumber());
+                snsClient.optInPhoneNumber(optInRequest);
+            }
+        }
+
```

Comment:
Cold we throw all this code into SmsService?

Reply:
I mean, I could, but at the time I wrote this, SmsService didn't exist yet, so...

Reply:
Moved to SmsService.